towerlib/towerlib.py

```diff
@@ -1997,24 +2034,707 @@ def schedules(self):
+        """Change the job template type in the given Ansible Tower, which means whether a job should run in 'Run' or
+        'Check' mode.
+
+        Args:
+            given_labels: ist of labels which need to match
+            new_job_type: the updated job type.
+
+        Returns:
+
+        """
+        count = 0
+        job_templates = self.get_all_job_templates()
+        if job_templates is None:
+            self._logger.error("Error finding job templates.")
+        else:
+            for job_template in job_templates:
+                if job_template.labels['count'] > 0:
+                    label_results = job_template.labels['results']
+                    job_labels = [item['name'] for item in label_results]
+                    if set(given_labels) == set(job_labels):
+                        count += 1
+                        self._logger.info("Equal labels found for the job template '{}'".format(job_template.name))
+                        existing_job_type = job_template.job_type
+                        if existing_job_type.lower() == new_job_type.lower():
+                            self._logger.info(
+                                "Existing job-type is already: '{}'. Change is not required for this job template.".
+                                    format(existing_job_type))
+                        else:
+                            self._logger.debug("Changing job type from '{}' to '{}'"
+                                               .format(existing_job_type, new_job_type))
+                            job_template_data = {
+                                "id": job_template.id,
+                                "job_type": new_job_type,
+                                "name": job_template.name,
+                                "playbook": job_template.playbook
+                            }
+                            self.change_job_template_data(job_template_data)
+            if count < 1:
+                self._logger.debug("No job template, which matched all the labels in the config file.")
+
+    def project_exists(self, project_name):
+        """Check if a project with given name exists in the project list.
+
+        Args:
+            project_name: the name of the given project.
+
+        Returns:
+            bool: True or False whether a project exists.
+
+            """
+        project_exists = False
+        projects = self.get_all_projects()
+        for project in projects:
+            if project.name == project_name:
+                project_exists = True
+        return project_exists
+
+    def get_all_credential_types(self):
+        """Get all the credential types of the ansible tower.
+
+        Returns:
+            list: list of credential types.
+
+        """
+        credential_types = []
+        for item in self.credential_types:
+            credential_types.append(item)
+        return credential_types
+
+    def get_credential_id_from_existing_project(self, scm_url):
```

Comment:
The name of the function does not reflect its usage. 
```get_credential_id_from_existing_project_by_scm_url``` is semantically more correct but it is a bit too long...